specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-privatepreview/getCachedServerName.json

```diff
@@ -0,0 +1,194 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationNameParameter"
+          },
+          {
+            "name": "cachedServerNameRequest",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CachedServerNameRequest"
+            },
+            "description": "The required parameters for get cached name available for fast provisioning."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CachedServerName"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CachedServerNameRequest": {
+      "type": "object",
+      "required": [
+        "version",
+        "storage",
+        "sku"
+      ],
+      "properties": {
+        "version": {
+          "$ref": "#/definitions/ServerVersion",
+          "description": "PostgreSQL Server version."
+        },
+        "storage": {
+          "$ref": "#/definitions/Storage",
+          "default": null,
+          "description": "Storage properties of a server."
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU (pricing tier) of the server."
+        }
+      },
+      "description": "Request from client to get available cached server name"
+    },
+    "CachedServerName": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of available cached server",
+          "readOnly": true
+        }
+      },
+      "description": "Represents a resource name of a cached server"
+    },
+    "ServerVersion": {
+      "type": "string",
+      "description": "The version of a server.",
+      "enum": [
+        "13",
+        "12",
+        "11"
+      ],
+      "x-ms-enum": {
+        "name": "ServerVersion",
+        "modelAsString": true
+      }
+    },
+    "Storage": {
+      "properties": {
+        "storageSizeGB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Max storage allowed for a server."
+        }
+      },
+      "description": "Storage properties of a server"
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the sku, typically, tier + family + cores, e.g. Standard_D4s_v3."
```

Comment:
Consider making the sku name as enum as well.

Reply:
Thanks, this was to follow the existing SKU properties in the rest of the files. From the next version and addition, we will follow your advice.